src/index.ts

```diff
@@ -203,7 +203,23 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
       })
     } else {
       headers.set('CF-Cache-Status', 'HIT')
-      response = new Response(response.body, { headers })
+      // fixes #165
+      let opts = {
+        headers,
+        status: 0,
+        statusText: ''
+      }
+      if (response.status) {
+        opts.status = response.status
+        opts.statusText = response.statusText
```

Comment:
i don't think you need to set the text, it will be probably generated later

Reply:
The answer is yes and no. According to the Fetch spec (at least at the time that we implemented it), the default status text is "OK" for all response status codes. We thought that was a bad default, so we deviated from the spec and codified a set of defaults based on http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml.

So, I think it's good practice to set the statusText for portable code that might also need to run in a browser, but for code that will only run on Cloudflare Workers, it can be safely omitted.

In this particular case, though, I think it's smart to preserve whatever our cache set.

Reply:
Good point, yes.